index.ts

```diff
@@ -257,11 +258,25 @@ export const exportEvents: BigQueryPlugin['exportEvents'] = async (events, { glo
             )
         }
 
-    } catch (error) {
+    } catch (error: any) {
         console.error(
             `Error inserting ${events.length} ${events.length > 1 ? 'events' : 'event'} into BigQuery: `,
             error
         )
+        // :TRICKY: If the message is too large for a single request, split it in half and try again
+        if (error?.message?.includes('Request Entity Too Large')) {
+            if (events.length > 1) {
+                const halfLength = Math.floor(events.length / 2)
+                await Promise.all([
```

Comment:
In that case it might be helpful to log that if one side failed/succeeded (but I'm guessing that logging will already come from the sub-calls, but would we know which side succeeded/failed). Use case: imagine I want to retry, it would be helpful to know that events between X-Y failed and Y-Z succeeded, so I can trigger a retry for just Y-Z.

Reply:
Seems you have a concrete, but hard-to-communicate idea in mind here. If it doesn't require changing the code too much, want to just take this over?

Reply:
Since it wasn't clear it's not important ;) We can add that later as needed.